actionview/lib/action_view/dependency_tracker/ripper_tracker.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module ActionView
+  class DependencyTracker # :nodoc:
+    class RipperTracker # :nodoc:
+      EXPLICIT_DEPENDENCY = /# Template Dependency: (\S+)/
+
+      def self.call(name, template, view_paths = nil)
+        new(name, template, view_paths).dependencies
+      end
+
+      def dependencies
+        render_dependencies + explicit_dependencies
+      end
+
+      def self.supports_view_paths? # :nodoc:
+        true
+      end
+
+      def initialize(name, template, view_paths = nil)
+        @name, @template, @view_paths = name, template, view_paths
+      end
+
+      private
+        attr_reader :template, :name, :view_paths
+
+        def render_dependencies
+          return [] unless template.source.include?("render")
+
+          compiled_source = template.handler.call(template, template.source)
```

Comment:
Are there performance implications in compiling the template's output here? Or to put it another way; what are the benefits in compiling here?

Reply:
There are performance implications, but it should be worth it. Most of this work is already done at boot time. One advantage to structuring the tracker this way is that it should work for all render handlers/types since Ripper can treat them all the same after they are compiled. This should mean slim, haml and any other format you might want will work accurately without modification. The most important benefit is increased accuracy. There are commented out renders and there are other ways ERBTracker could be mistakes about what the dependencies of a template are. This should avoid those problems.

Reply:
> Most of this work is already done at boot time.

That's true in https://github.com/jhawthorn/actionview_precompiler but is it true here? Won't the files be lazy-loaded when a cache block is encountered (and a digest is needed) during rendering, as opposed to being precompiled at boot time?

Reply:
>There are performance implications, but it should be worth it. 

Is it worth doing a bit of profiling and benchmarking?

Reply:
@Matt-Yorkley  I think you are right about that case for the first time that that template is read or when the details change. After that, Rails should already know the digest. I am going to work on some benchmarks anyway since I should have posted those before regardless.

We should also be ready to move more template digesting to boot time now which should be better regardless since that should be faster no matter how much slower ripper is compared to ERB tracker.